homeassistant/components/ps4/__init__.py

```diff
@@ -31,3 +33,40 @@
     await hass.config_entries.async_forward_entry_unload(
         entry, 'media_player')
     return True
+
+
+async def async_migrate_entry(hass, entry):
+    """Migrate old entry."""
+    from pyps4_homeassistant.media_art import COUNTRIES
+
+    config_entries = hass.config_entries
+    data = entry.data
+    version = entry.version
+
+    reason = {1: "Region codes have changed"}
+
+    # Migrate Version 1 -> Version 2
+    if version == 1:
+        loc = location.detect_location_info()
+        if loc:
+            country = loc.country_name
+            if country in COUNTRIES:
+                for device in data['devices']:
+                    device[CONF_REGION] = country
+                entry.version = 2
+                config_entries.async_update_entry(entry, data=data)
+                _LOGGER.info(
+                    "PlayStation 4 Config Updated: \
+                    Region changed to: %s", country)
+                return True
+
+    msg = """{} for the PlayStation 4 Integration.
+            Please remove the PS4 Integration and re-configure
+            [here](/config/integrations).""".format(reason[version])
+
+    hass.components.persistent_notification.async_create(
```

Comment:
If config entry version is 2, we shouldn't show the notification, right? Or are only non latest version migrated?

Reply:
If version 2 is the latest, async_migrate_entry() never gets called right? My intent was for if the version is current there would be no notifications. I put this as a template for future versions so it should be non-latest versions are migrated when a new version is added.

Reply:
I'm not sure. The entry migration code is very new. Did you look at that or test that's how it works?

Reply:
Ok, I checked. It's like you say, if the version of the handler is the same as the version of the entry, the component migration function isn't called.